_ml-commons-plugin/conversational-search.md

```diff
@@ -0,0 +1,367 @@
+        "model_id": "<model_id>",
+        "context_field_list": ["text"]
+      }
+    }
+  ]
+}
+```
+
+`context_field_list` is the list of fields in document sources that the pipeline uses as context for the RAG. For example, when `context_field_list` parses through the following document:
+
+```json
+{
+  "_index": "qa_demo",
+  "_id": "SimKcIoBOVKVCYpk1IL-",
+  "_source": {
+    "title": "Abraham Lincoln 2",
+    "text": "Abraham Lincoln was born on February 12, 1809, the second child of Thomas Lincoln and Nancy Hanks Lincoln, in a log cabin on Sinking Spring Farm near Hodgenville, Kentucky.[2] He was a descendant of Samuel Lincoln, an Englishman who migrated from Hingham, Norfolk, to its namesake, Hingham, Massachusetts, in 1638. The family then migrated west, passing through New Jersey, Pennsylvania, and Virginia.[3] Lincoln was also a descendant of the Harrison family of Virginia; his paternal grandfather and namesake, Captain Abraham Lincoln and wife Bathsheba (née Herring) moved the family from Virginia to Jefferson County, Kentucky.[b] The captain was killed in an Indian raid in 1786.[5] His children, including eight-year-old Thomas, Abraham's father, witnessed the attack.[6][c] Thomas then worked at odd jobs in Kentucky and Tennessee before the family settled in Hardin County, Kentucky, in the early 1800s.[6]\n"
+  }
+}
+```
+
+When configured using the preceeding Search API request, the pipeline sends the `text` field from the response to OpenAI model. You can customize `context_field_list`  in your RAG pipeline to send any fields that exist in your documents to the LLM.
+
+### Using the pipeline
+
+Using the pipeline is similar to submitting [search queries]({{site.url}}{{site.baseurl}}/api-reference/search/#example) to OpenSearch, as shown in the following example:
+
+```json
+GET /<index_name>/_search?search_pipeline=<pipeline_name>
+{
+	"query" : {...},
+	"ext": {
+		"generative_qa_parameters": {
+			"llm_model": "gpt-3.5-turbo",
+			"llm_question": "Was Abraham Lincoln a good politician",
+			"conversation_id": "_ikaSooBHvd8_FqDUOjZ"
+		}
+	}
+}
+```
+
+The RAG search query uses the following request objects under the `generative_qa_paramters` option:
+
+Parameter | Required | Description
+:--- | :--- | :---
+`llm_question` | Yes | The question the LLM must answer. 
+`llm_model` | No | Overrides the original model set in the connection in cases where you want to use a different model. For example, using GPT 4 instead of GPT 3.5. This option is required if a default model is not set during pipeline creation.
+`coversation_id` | No | Integrates conversation memory into your RAG pipeline by adding the 10 most recent conversations into the context of search query to the LLM. 
+
+If you're LLM includes a set token limit, set the `size` field in you OpenSearch query to limit the amount of documents used in the search response. Otherwise, the RAG pipeline will send every document in the search result into the LLM.
```

Comment:
```suggestion
If your LLM includes a set token limit, set the `size` field in your OpenSearch query to limit the number of documents used in the search response. Otherwise, the RAG pipeline will send every document in the search results to the LLM.
```